File: client/src/models/application.ts
import type { ApplicationStatus } from './enums';
import type { Job } from './job';

export interface JobApplication {
  id: string;
  status: ApplicationStatus;
  appliedAt: string;
  notes?: string;
  userProfileId: string;
  jobId: string;
  job?: Job;
}

export interface CreateApplicationRequest {
  jobId: string;
}

File: server/src/SwipeJobs.Domain/Entities/Application.cs
using SwipeJobs.Domain.Common;
using SwipeJobs.Domain.Enums;

namespace SwipeJobs.Domain.Entities;

public class Application : BaseEntity
{
    public ApplicationStatus Status { get; set; } = ApplicationStatus.Pending;
    public DateTime AppliedAt { get; set; } = DateTime.UtcNow;
    public string? Notes { get; set; }

    public Guid UserProfileId { get; set; }
    public UserProfile UserProfile { get; set; } = null!;

    public Guid JobId { get; set; }
    public Job Job { get; set; } = null!;
}
